File: sdk/python/hitechcloud/resources/pmg.py
"""HiTechCloudPMG resource (4 endpoints)"""

from .base import BaseResource

class PmgResource(BaseResource):
    """HiTechCloudPMG: mail filtering configuration and management"""

    def get_config(self, service_id: int) -> dict:
        """GET /api/service/{id}/htcpmg/config - Get mail filtering configuration"""
        return self._http.get(f"/api/service/{service_id}/htcpmg/config")

    def add_domain(self, service_id: int, domain: str) -> dict:
        """POST /api/service/{id}/htcpmg/domains - Add a domain"""
        return self._http.post(f"/api/service/{service_id}/htcpmg/domains", {"domain": domain})

    def set_transport(self, service_id: int, domain: str, target: str) -> dict:
        """POST /api/service/{id}/htcpmg/transport - Set target mail server"""
        return self._http.post(f"/api/service/{service_id}/htcpmg/transport", {"domain": domain, "target": target})

    def get_stats(self, service_id: int) -> dict:
        """GET /api/service/{id}/htcpmg/stats - Get mail statistics"""
        return self._http.get(f"/api/service/{service_id}/htcpmg/stats")

File: sdk/ruby/lib/hitechcloud/resources/pmg.rb
module HiTechCloud
  module Resources
    class Pmg < Base

        # Get config
        def config(id)
          @http.get("/api/service/%{id}/htcpmg/config" % { id: id })
        end

        # Add domain
        def add_domain(id, data)
          @http.post("/api/service/%{id}/htcpmg/domains" % { id: id }, data)
        end

        # Add transport
        def add_transport(id, data)
          @http.post("/api/service/%{id}/htcpmg/transport" % { id: id }, data)
        end

        # Get stats
        def stats(id)
          @http.get("/api/service/%{id}/htcpmg/stats" % { id: id })
      end
    end
  end
end
